File: src/RealWorld.Core/Entities/ArticleFavorite.cs
namespace RealWorld.Core.Entities;

public class ArticleFavorite
{
    public string ArticleId { get; set; } = string.Empty;
    public string UserId { get; set; } = string.Empty;

    public ArticleFavorite() { }

    public ArticleFavorite(string articleId, string userId)
    {
        ArticleId = articleId;
        UserId = userId;
    }
}

File: src/main/java/io/spring/core/favorite/ArticleFavorite.java
package io.spring.core.favorite;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * Value object representing a user's favorite relationship with an article.
 *
 * <p>Composite key of {@code articleId} and {@code userId}.
 */
@NoArgsConstructor
@Getter
@EqualsAndHashCode
public class ArticleFavorite {
  private String articleId;
  private String userId;

  public ArticleFavorite(String articleId, String userId) {
    this.articleId = articleId;
    this.userId = userId;
  }
}
